cmd/botkube/main.go

```diff
@@ -388,7 +388,7 @@ func run(ctx context.Context) error {
 		}()
 		heartbeatReporter := heartbeat.GetReporter(remoteCfgEnabled, logger, gqlClient)
 		k8sCollector := insights.NewK8sCollector(k8sCli, heartbeatReporter, logger, reportHeartbeatInterval, reportHeartbeatMaxRetries)
-		return k8sCollector.Start(ctx)
```

Comment:
why not to simply do the 
```go
	if !remoteCfgEnabled {
		k.logger.Debug("Remote config is not enabled, skipping k8s insights collection...")
		return nil
	}
```

at the beginning of the func? As a result also the `heartbeatReporter` and `k8sCollector` won't be created at all. Also the  `heartbeat.GetReporter` could be simplified.

Reply:
How can't I see that :) on it...

Reply:
Updated, please check

Reply:
@mszostok as you suggested, I also simplified reporter